tests/src/TabulatedPolynomialModel.cxx

```diff
@@ -0,0 +1,178 @@
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "TabulatedPolynomialModel.h"
+
+using namespace neml2;
+
+register_NEML2_object(TabulatedPolynomialModel);
+
+OptionSet
+TabulatedPolynomialModel::expected_options()
+{
+  auto options = Model::expected_options();
+  options.set<VariableName>("von_mises_stress") = VariableName("state", "s");
+  options.set<VariableName>("temperature") = VariableName("forces", "T");
+  options.set<VariableName>("internal_state_1") = VariableName("state", "s1");
+  options.set<VariableName>("internal_state_2") = VariableName("state", "s2");
+  options.set<VariableName>("equivalent_plastic_strain_rate") = VariableName("state", "ep_rate");
+  options.set<VariableName>("internal_state_1_rate") = VariableName("state", "s1_rate");
+  options.set<VariableName>("internal_state_2_rate") = VariableName("state", "s2_rate");
+  options.set<Real>("index_sharpness") = 1.0;
+  return options;
+}
+
+TabulatedPolynomialModel::TabulatedPolynomialModel(const OptionSet & options)
+  : Model(options),
+    _s(declare_input_variable<Scalar>("von_mises_stress")),
+    _T(declare_input_variable<Scalar>("temperature")),
+    _s1(declare_input_variable<Scalar>("internal_state_1")),
+    _s2(declare_input_variable<Scalar>("internal_state_2")),
+    _ep_dot(declare_output_variable<Scalar>("equivalent_plastic_strain_rate")),
+    _s1_dot(declare_output_variable<Scalar>("internal_state_1_rate")),
+    _s2_dot(declare_output_variable<Scalar>("internal_state_2_rate")),
+    _A0(declare_buffer<BatchTensor>("A0", A0())),
+    _A1(declare_buffer<BatchTensor>("A1", A1())),
+    _A2(declare_buffer<BatchTensor>("A2", A2())),
+    _s_lb(declare_buffer<BatchTensor>("s_lb", s_lb())),
+    _s_ub(declare_buffer<BatchTensor>("s_ub", s_ub())),
+    _T_lb(declare_buffer<BatchTensor>("T_lb", T_lb())),
+    _T_ub(declare_buffer<BatchTensor>("T_ub", T_ub())),
+    _k(options.get<Real>("index_sharpness"))
+{
+}
+
+BatchTensor
```

Comment:
If we're doing to merge this we should make the coefficient matrices be user input.  You should be able to just make these User tensors, right?